content/docs/reference-react.md

```diff
@@ -103,16 +103,14 @@ React.cloneElement(
 )
 ```
 
-Clone and return a new React element using `element` as the starting point. The resulting element will have the original element's props with the new props merged in shallowly. New children will replace existing children. `key` and `ref` from the original element will be preserved.
+Clone and return a new React element using `element` as the starting point. The resulting element will have the original element's props with the new props merged in shallowly. New children, if passed as `...children` or otherwise `props.children`, will replace existing children. `key` and `ref` from the original element will be preserved unless `props` contains `key` or `ref` (which will overwrite the `key` or `ref` in the cloned element, respectively).
 
 `React.cloneElement()` is almost equivalent to:
 
 ```js
 <element.type {...element.props} {...props}>{children}</element.type>
 ```
 
-However, it also preserves `ref`s. This means that if you get a child with a `ref` on it, you won't accidentally steal it from your ancestor. You will get the same `ref` attached to your new element.
```

Comment:
I think this is a useful sentence. Maybe we could just tweak the wording slightly?

> Except that it also preserves `ref`s by default. This means that if you get a child with a `ref` on it, you won't accidentally steal it from your ancestor. You will get the same `ref` attached to your new element.

Reply:
I think this sentence is very unclear though.  Is "child with a `ref` on it" referring to a child of the cloned element or the cloned element itself?

Reply:
You could replace, "child with a `ref` on it" with something more like, "if you clone an element that has a `ref`".

Reply:
but up above the code example it already says "`key` and `ref` from the original element will be preserved."